### PR TITLE
Fix no-assert code-smell regex.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Fix no-assert code-smell regex.

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

no-assert sanity test

### CODE REVIEW

File: test/sanity/code-smell/no-assert.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import re
 import sys
 
-ASSERT_RE = re.compile(r'.*(?<![-:a-zA-Z#][ -])\bassert\b(?!:).*')
+ASSERT_RE = re.compile(r'^\s*assert[^a-z0-9_:]')
```
Comment (bcoca) [inline test/sanity/code-smell/no-assert.py:8]: won't that allow for a expression like `stuff or assert ...` ?

Comment (mattclay) [reply test/sanity/code-smell/no-assert.py:8]: Yes, it will. Thankfully that isn't valid Python syntax.

Comment (bcoca) [reply test/sanity/code-smell/no-assert.py:8]: ah, TIL

**Comment by sivel** [comment]
I think this is probably sufficient.  I think the previous regex was likely over complicated, and I don't have good reasons for why it ended up that way.  This will likely speed things up too, as the old regex was fairly slow.

**Review by sivel — APPROVED** [review]

**Pull request merged**